### DISCUSSION TITLE
Error Fetching: {"code": "22P02", "details": null, "hint": null, "message": "invalid input syntax for type uuid: \"\""}

### DISCUSSION THREAD

**Discussion by CraigStueber** [supabase/supabase#22480] [Questions]
I keep getting this error code when I am fetching my user. the uuid is created via Supabase

**Comment by GaryAustin1** [comment]
You need to provide more info on the actual call you are doing.  What is fetching my user?

**Reply by CraigStueber** [reply]
Okay no worries I am fetching or running an async function to 


  const [userID, setUserID] = useState("");
  const [userProfile, setUserProfile] = useState("");

  useEffect(() => {
    const getUser = async () => {
      const user = await fetchUserProfile();
      setUserProfile(user);
    };
    getUser();
  }, [userID]);

  async function fetchUserProfile() {
    setUserID(props.session.user.id);
    let { data: UserProfile, error } = await supabase
      .from("UserProfile")
      .select("*")
      .eq("UserID", userID);
    if (error) {
      console.error("Error fetching: ", error);

      return;
    } else {
      return UserProfile;
    }
  }

I am receiving this Data package

[{"Activities": ["Warhammer", "Coding", "Board Games", "Boxing", "Picking On Scott"], "City": "Richmond", 
"Description": "something",
"FirstName": "Craig", "LastName": "Stueber", "State": "VA", "UserID": "bd0597d3-82bc-4d60-9043-8892afd74150", "UserName": "JAZZAAR", "avatar": "craigAvatar.jpg", "created_at": "2023-12-14T23:53:07+00:00", "id": 1, "varPhoto": "craigVar.jpg"}]
Getting this Error:
 ERROR  Error fetching:  {"code": "22P02", "details": null, "hint": null, "message": "invalid input syntax for type uuid: \"\""}
 
 The UUID is only the UserID which is automatically given by Supabase. but the error is saying it is invalid. so I am confused. I am pretty sure it is just some error I did in my code

**Reply by GaryAustin1** [reply]
I would console.log userID and make sure it looks correct (should be a string with no extra quotes and not null). 

You could also hardcode UUID in the .eq() to check which side the error is on.

**Reply by CraigStueber** [reply]
LOG  bd0597d3-82bc-4d60-9043-8892afd74150
 
 Is what I am getting from console.log

**Reply by CraigStueber** [reply]
I hard coded the UUID into the eq() 

 .eq("UserID", "bd0597d3-82bc-4d60-9043-8892afd74150");
 
 And the error is still there

**Reply by GaryAustin1** [reply]
Sort of puzzling.  I don't think the error is this, but make sure UserID is a UUID column.
Also if this is next.js it does cache selects in many cases.   Try changing .select('*') to .select('userID') which would bust the cache if that was it.

You could also drop the .eq() as a test to make sure it is the filter causing the error.  Long shot.

**Reply by CraigStueber** [reply]
Okay so I now see I am getting this error from only when pulling a user that is using the UUID from Supabase 

I added in the console.error message "Error fetching: user" "Error fetching: happening" or what ever it is going to get. 

The error I am getting is always from the user.

**Reply by CraigStueber** [reply]
It is something do with the eq
As if I make that go away and filter the query in js it completely drops the error 

So I need to figure out what I have done wrong :) which is half of our life after all :)

**Reply by CraigStueber** [reply]
What is puzzling too is the function is doing its job. It is sending me the user I am looking for. I am thinking about just since we are not in deployment and just in prototyping. If/elsing this error out of showing up and keep on and doing some research on it later

**Reply by GaryAustin1** [reply]
It could be it got cached in a bad state by next.js.  Dropping the eq changed the url.  You could also try a different UUID.  That would bust the cache also. 

Edit I have low confidence in this being the issue based on what you are seeing.

**Reply by robksawyer** [reply]
Any luck with this? I'm running into the same thing when doing queries such as: 
```
const { data: userRoles, error: rolesError } = await supabase
      .from("user_roles")
      .select(
        `
        role_id,
        roles!fk_user_roles_roles (
          name,
          role_permissions (
            permissions (
              name
            )
          )
        )
      `
      )
      .eq("user_id", userId);
``` 

in NextJS 14.2.4. When I encodeURIComponent on the query, it returns data, but only the user_roles table data and not the joins or other data that I need.
